tests/baselines/reference/dependentReturnType1.errors.txt

```diff
@@ -0,0 +1,645 @@
+            }
+            this.root.name = name;
+            return name; // Error
+            ~~~~~~
+!!! error TS2322: Type 'T & {}' is not assignable to type 'this'.
+!!! error TS2322:   'this' could be instantiated with an arbitrary type which could be unrelated to 'T & {}'.
+        }
+    }
+    
+    // Conditional expressions
+    interface Aa {
+        1: number;
+        2: string;
+        3: boolean;
+    }
+    
+    function trivialConditional<T extends 1 | 2 | 3>(x: T): Aa[T] {
+        if (x !== 1) {
+            return x === 2 ? "" : true;
+        }
+        else {
+            return 0;
+        }
+    }
+    
+    function conditional<T extends boolean>(x: T):
+     T extends true ? 1 : T extends false ? 2 : never {
+        return x ? 1 : 2; // Ok
+    }
+    
+    function contextualConditional<T extends "a" | "b">(
+        x: T
+    ): T extends "a" ? "a" : T extends "b" ? number : never {
+        return x === "a" ? x : parseInt(x); // Ok
+    }
+    
+    function conditionalWithError<T extends "a" | "b">(
+        x: T
+    ): T extends "a" ? number : T extends "b" ? string : never {
+        return x === "a" ? x : parseInt(x); // Error
+                           ~
+!!! error TS2322: Type 'string' is not assignable to type 'number'.
+                               ~~~~~~~~~~~
+!!! error TS2322: Type 'number' is not assignable to type 'string'.
+    }
+    
+    // Multiple indexed type reductions
+    interface BB {
+        "a": number;
+        [y: number]: string;
+    }
+    
+    interface AA<T extends keyof BB> {
+        "c": BB[T];
+        "d": boolean,
+    }
+    
+    function reduction<T extends keyof BB, U extends "c" | "d">(x: T, y: U): AA<T>[U] {
+        if (y === "c" && x === "a") {
+            // AA<T>[U='c'] -> BB[T]
+            // BB[T='a'] -> number
+            return 0; // Ok
+        }
+    
+        return undefined as never;
+    }
+    
+    // Substitution types are not narrowed
+    function subsCond<T extends 1 | 2 | 3>(
+        x: T,
+    ): T extends 1 | 2
+        ? T extends 1
+            ? string
+            : T extends 2
+              ? boolean
+              : never
+        : T extends 3
+          ? number
+          : never {
+        if (x === 1) {
+            return "";
+            ~~~~~~
+!!! error TS2322: Type '""' is not assignable to type 'T extends 1 | 2 ? T extends 1 ? string : T extends 2 ? boolean : never : T extends 3 ? number : never'.
+        } else if (x == 2) {
+            return true;
+            ~~~~~~
+!!! error TS2322: Type 'true' is not assignable to type 'T extends 1 | 2 ? T extends 1 ? string : T extends 2 ? boolean : never : T extends 3 ? number : never'.
+        }
+        return 3;
+        ~~~~~~
+!!! error TS2322: Type '3' is not assignable to type 'T extends 1 | 2 ? T extends 1 ? string : T extends 2 ? boolean : never : T extends 3 ? number : never'.
+    }
+    
+    
+    // Unsafe: check types overlap

```

Comment:
why is this unsafe? Is it because objects of type T could have additional properties that aren't represented in the type?

Reply:
oh, I didn't read all the way. Yes, that is it.